ci/infra/testrunner/tests/test_istio_service_mesh.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import pytest
+import time
+
+CLUSTERROLEBINDING = ("""
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: istio-test
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: suse:caasp:psp:privileged
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: default
+- kind: ServiceAccount
+  name: httpbin
+  namespace: default
+- kind: ServiceAccount
+  name: bookinfo-productpage
+  namespace: default
+- kind: ServiceAccount
+  name: bookinfo-reviews
+  namespace: default
+- kind: ServiceAccount
+  name: bookinfo-ratings 
+  namespace: default
+- kind: ServiceAccount
+  name: bookinfo-details
+  namespace: default
+- kind: ServiceAccount
+  name: sleep
+  namespace: default
+EOF
+""")
+
+
+def _istio_bookinfo_setup(kubectl):
+    istioctl = ("""
+                istioctl --kubeconfig={config} manifest apply \
+                         --set profile=default \
+                         --set addonComponents.prometheus.enabled=false \
+                         --set hub=registry.suse.de/devel/caasp/4.5/containers/containers/caasp/v4.5 \
+                         --set tag=1.5.4 \
+                         --set values.pilot.image=istio-pilot \
+                         --set values.global.proxy.image=istio-proxyv2 \
+                         --set values.global.proxy_init.image=istio-proxyv2
+                 """.format(config=kubectl.get_kubeconfig()))
+
+    kubectl.utils.runshellcommand(istioctl)
+    kubectl.run_kubectl("-n istio-system wait --for=condition=available deploy/istio-ingressgateway --timeout=3m")
+
+    # Activate service in namespace default
+    kubectl.run_kubectl("label namespace default istio-injection=enabled --overwrite")
+
+    # Create the clusterrolebinding to run everything
+    kubectl.run_kubectl("apply -f - << EOF " + CLUSTERROLEBINDING)
+
+    # Deploy bookinfo application
+    kubectl.run_kubectl("apply -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/platform/kube/bookinfo.yaml")
+    kubectl.run_kubectl("-n default wait --for=condition=available deploy/productpage-v1 --timeout=3m")
+
+    # Deploy the gateway and destination rules
+    kubectl.run_kubectl("apply -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/networking/bookinfo-gateway.yaml")
+    kubectl.run_kubectl("apply -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/networking/destination-rule-all.yaml")
+
+    # Deploy the sleep pod that we use as client
+    kubectl.run_kubectl("apply -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/sleep/sleep.yaml")
+    kubectl.run_kubectl("-n default wait --for=condition=available deploy/sleep --timeout=3m")
+
+
+def _cleanup(kubectl):
+    kubectl.run_kubectl("delete -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/platform/kube/bookinfo.yaml")
```

Comment:
I would put the fixed part of the URL into a variable, to make the code more readeble. Also, it would make it easier to change the version and makes the code easier to maintain.

Reply:
Very good point. I'll fix that in a minute